spec/overcommit/hook_context/prepare_commit_msg_spec.rb

```diff
@@ -0,0 +1,38 @@
+require 'spec_helper'
+require 'overcommit/hook_context/prepare_commit_msg'
+
+describe Overcommit::HookContext::PrepareCommitMsg do
+  let(:config) { double('config') }
+  let(:args) { [commit_msg_filename, commit_msg_source] }
+  let(:commit_msg_filename) { 'message-template.txt' }
+  let(:commit_msg_source) { :file }
+  let(:commit) { 'SHA-1 here' }
+  let(:input) { double('input') }
+  let(:context) { described_class.new(config, args, input) }
+
+  describe '#commit_msg_filename' do
+    subject { context.commit_msg_filename }
+
+    it { should == commit_msg_filename }
+  end
+
+  describe '#commit_msg_source' do
+    subject { context.commit_msg_source }
+
+    it { should == commit_msg_source }
+  end
+
+  describe '#commit' do
+    subject { context.commit }
+
+    context "source isn't :commit" do
+      it { should == `git rev-parse HEAD` }
```

Comment:
This is a bit ugly. Can you think of a better way to write this spec?

Reply:
You'll be able to remove this particular test given my comment above.